src/cargo/ops/cargo_new.rs

```diff
@@ -866,14 +867,40 @@ mod tests {
             }
         }
     }
-
-    if let Err(e) = Workspace::new(&path.join("Cargo.toml"), config) {
-        crate::display_warning_with_error(
-            "compiling this new package may not work due to invalid \
+    match Workspace::new(&path.join("Cargo.toml"), config) {
```

Comment:
> Doing a single write would likely be more complicated on the workspace side (slightly, I suspect we can just reuse "find workspace" function that I think exists) but seems like it'd be easier to format it as we wish vs editing it afterwards (even if editing is easier because we can assume what we are editing.

Yes. I agree. I will try `find_workspace_root `.

Reply:
But we also have another issue about making sure it is a part of the workspace member.  Is this approach throwaway if we do that?  If so, I'd rather we just do this right rather  than hack something together and then have to go through this process all over again.

Reply:
> Is this approach throwaway if we do that? If so, I'd rather we just do this right rather than hack something together and then have to go through this process all over again.

The function is extracted. So it is very easy to call it after we solve that issue.